tests/adapter/dbt/tests/adapter/test_incremental_unique_id.py

```diff
@@ -0,0 +1,613 @@
+import pytest
+from dbt.tests.util import run_dbt
+from dbt.tests.tables import TableComparison
+from dbt.contracts.results import RunStatus
+from collections import namedtuple
+from pathlib import Path
+
+models__trinary_unique_key_list_sql = """
+-- a multi-argument unique key list should see overwriting on rows in the model
+--   where all unique key fields apply
+
+{{
+    config(
+        materialized='incremental',
+        unique_key=['state', 'county', 'city']
+    )
+}}
+
+select
+    state::varchar(2) as state,
+    county::varchar(12) as county,
+    city::varchar(12) as city,
```

Comment:
We'll want to be thoughtful about data types in order to get this running successfully on multiple databases. E.g. BigQuery doesn't support a data type named `varchar(length)`. Our options here:
- Override/reimplement all fixtures when we run this test in `dbt-bigquery` (ugh)
- Remove type casts here, try to handle all data type business via seeds instead. That may also require a different approach for `add_new_rows.sql`/`duplicate_insert.sql`

Reply:
These tests are also working on the adapter repos? Did you change the datatypes?

Reply:
We could provide some of that information (with data_types) in a custom fixture. Avoiding that would be even better. There is actually an update_rows function in dbt.tests.util which came from dbt-adapter-tests and does work on all of our repos because it calls an adapter method.

Also, I think putting the test settings in variables and then putting them into the structures isn't the best practice. I'm gong to look at it a bit more to come up with recommendations.

Reply:
> These tests are also working on the adapter repos? Did you change the datatype

Yes, I believe there are slight modifications for each adapter repo

> There is actually an update_rows function in dbt.tests.util which came from dbt-adapter-tests and does work on all of our repos because it calls an adapter method.

I'm a bit skeptical of this method tbh (I know it required us to override a data type in Redshift for basic tests), and want to discuss alongside table comparison logic—but maybe it's the way to go!

Maybe the move should be:
- Once this test case is passing in this repo (for Postgres), we try running it (from this branch) in a branch of each plugin repo as well
- Update the test cases here, override them there, get them all passing for all plugins
- Fine-tune the framework so that the overrides feel minimal + appropriate
- Merge across the board (and delete all the previously duplicated test code)

Reply:
That sounds like the right process.

The override in Redshift for a seed data column was because the test goes and updates the name by appending '_update'. Redshift created the seed table with a varchar datatype that wasn't big enough. The best solution might actually have been to change that update so the size didn't increase, but the 'update_rows' method was kind of fiddly.